cypress/integration/api/alerts/device_alerts.ts

```diff
@@ -0,0 +1,137 @@
+import moment from "moment";
+import { EventTypes } from "../../../commands/api/events";
+import { checkResponse, getCreds } from "../../../commands/server";
+import { getTestName } from "../../../commands/names";
+import { ComparableAlert, getExpectedAlert } from "../../../commands/api/alerts";
+import { ComparableEvent, getExpectedEvent } from "../../../commands/api/events";
+
+const AuthorizationError = 403;
+const BadRequest = 400;
+
+const OK = null;
+
+
+describe("Devices alerts", () => {
+  const group = "colonelGroup";
+  const user = "colonel";
+  const helper = 'frank';
+  const camera = 'camera1';
+  const conditions = [{"tag":"possum","automatic":true}];
+  const alertName = "alert1";
+  var oneAlert;
+  var emptyAlert;
+  var expectedEvent;
+  var alertId;
+  var recId;
+
+  before(() => {
+    cy.apiCreateUser(helper);
+    cy.apiCreateUserGroupAndCamera(user, group, camera);
+  });
+
+  it("Cannot create alert without access permissions", () => {
+    cy.apiAddAlert(helper,alertName,conditions,camera,null,AuthorizationError).then((response) => {checkResponse(response,AuthorizationError)});
+  });
+
+  it("Cannot create alert with invalid condition", () => { 
+    const bad_conditions = [{"tag": "any", "automaticF": true}]
+    cy.apiAddAlert(user,alertName,bad_conditions,camera,null,BadRequest).then((response) => {checkResponse(response,BadRequest)});
+  });
+
+  it("Can create alert and has no events by default", () => {
+    // create alert
+    cy.apiAddAlert(user,alertName,conditions,camera,0,200).then((response)=> {
+
+    // crete an example alert to compare against
+    cy.createExpectedAlert("emptyAlert",{"id": getCreds(alertName).id,
+        "name": "alert1",
+        "frequencySeconds": 0,
+        "conditions": [{"tag":"possum","automatic":true}],
+        "lastAlert":false,
+        "User":{"id":getCreds(user).id, "username":getTestName(getCreds(user).name), "email":getTestName(getCreds(user).name)+"@api.created.com"},
+        "Device":{"id":getCreds(camera).id, "devicename":getTestName(getCreds(camera).name)}});
+    });
+
+    //check we created an alert wth no events yet
+    cy.apiCheckAlert(user,camera,"emptyAlert");
+  });
+
+  it("Can receive an alert", () => {
+    //expected alert to compare against 
+    cy.createExpectedAlert("alert1",{"id": getCreds(alertName).id,
+    "name": "alert1",
+    "frequencySeconds": 0,
+    "conditions": [{"tag":"possum","automatic":true}],
+    "lastAlert":true,
+    "User":{"id":getCreds(user).id, "username":getTestName(getCreds(user).name), "email":getTestName(getCreds(user).name)+"@api.created.com"},
+    "Device":{"id":getCreds(camera).id, "devicename":getTestName(getCreds(camera).name)}});
+    
+    //upload a recording tagged as possum and  build an expected event using the returned recording details
+    cy.uploadRecording(camera, { tags: ["possum"] }, null, "recording1").then((response)=>{
+      cy.createExpectedEvent("event1",{"id":1,
```

Comment:
Instead of setting the expected event, can you just not do the apiCheckEvent from within the then statement?   That would get ride of this extra complicatedness of createExpectedEvent.